client/order/epochqueue.go

```diff
@@ -0,0 +1,190 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package order
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"sort"
+	"sync"
+
+	"decred.org/dcrdex/dex/msgjson"
+	"decred.org/dcrdex/dex/order"
+	"github.com/decred/dcrd/crypto/blake256"
+)
+
+// EpochQueue represents a client epoch queue.
+type EpochQueue struct {
+	orders             map[order.OrderID]*msgjson.EpochOrderNote
+	ordersMtx          sync.Mutex
+	seedIndex          []order.OrderID
+	seedIndexMtx       sync.Mutex
+	commitmentIndex    []order.OrderID
+	commitmentIndexMtx sync.Mutex
```

Comment:
Are there any cases where either of `seedIndexMtx` or `commitmentIndexMtx` would be locked but **not** `ordersMtx          sync`?  I suspect you could eliminate the two index mutexes.

This would simplify Reset as well.  e.g.

```go
func (eq *EpochQueue) Reset() {
	eq.ordersMtx.Lock()
	eq.orders = make(map[order.OrderID]*msgjson.EpochOrderNote)
	eq.seedIndex = nil
	eq.commitmentIndex = nil
	eq.ordersMtx.Unlock()
}
```